web/src/components/projects/CreateProjectForm.tsx

```diff
@@ -0,0 +1,179 @@
+import { useState } from "react";
+import { useForm } from "react-hook-form";
+import { postProjectsTre, ProjectTreRequest, ValidationError, Study } from "@/openapi";
+import Button from "../ui/Button";
+import Dialog from "../ui/Dialog";
+import { HelperText, Alert, AlertMessage } from "../shared/exports";
+
+import styles from "./CreateProjectForm.module.css";
+
+type ProjectFormData = {
+  name: string;
+  studyId: string;
+  environment: "tre" | "dsh";
+};
+
+type Props = {
+  studies: Study[];
+  handleProjectCreated: () => void;
+  handleCancelCreate: () => void;
+};
+
+export default function CreateProjectForm({ studies, handleProjectCreated, handleCancelCreate }: Props) {
+  const [isSubmitting, setIsSubmitting] = useState(false);
+  const [error, setError] = useState<string | null>(null);
+
+  const {
+    register,
+    handleSubmit,
+    formState: { errors },
+  } = useForm<ProjectFormData>({
+    defaultValues: {
+      name: "",
+      studyId: "",
+      environment: undefined,
+    },
+  });
+
+  const onSubmit = async (data: ProjectFormData) => {
+    setIsSubmitting(true);
+    setError(null);
+
+    try {
+      let response;
+
+      switch (data.environment) {
+        case "tre": {
+          const requestBody: ProjectTreRequest = {
+            name: data.name,
+            study_id: data.studyId,
+          };
+          response = await postProjectsTre({ body: requestBody });
+          break;
+        }
+        case "dsh":
+          throw new Error("DSH projects are not yet supported");
+        default:
+          throw new Error(`Unknown environment: ${data.environment}`);
+      }
```

Comment:
Depending on how many environments we end up with, we might want a neater solution to this. Maybe some sort of lookup that contains more project attributes for each environment.